R/scATAC.Peak.annotation.R

```diff
@@ -407,16 +420,17 @@ peaks_on_gene <- function(peak_features,annotations=NULL, gene_element=NULL, spl
     cat(paste("done", "time", difftime(end_time, start_time, units="secs"), "s", "\n", sep = " "))
   }
   peak_ongene <- do.call(rbind, peak_list)
+  rownames(peak_ongene) <- paste0(peak_ongene[,1],":", peak_ongene[,2], "-", peak_ongene[,3])
   if (TSSmode==T){
-    rownames(peak_ongene) <- paste0(peak_ongene[,1],":", peak_ongene[,2], "-", peak_ongene[,3])
-    colnames(peak_ongene) <- c("seqnames", "Pstart", "Pend", "peak_on_gene", "strand", "TSSinfo", "TSSdistance", "overlap.alter.TSS")
-    peaks_length <- apply(peak_ongene, 1, function(x) {
-      length(as.numeric(x[2]):as.numeric(x[3]))
-    })
-    peak_ongene<- cbind(peak_ongene, peaks_length=peaks_length)
+    colnames(peak_ongene) <- c("seqnames", "Pstart", "Pend", "gene_id", "peak_on_gene", "strand", "TSSinfo", "TSSdistance", "overlap.alter.TSS")
   }
   else{
-    rownames(peak_ongene) <-  paste0(peak_ongene[,1],":", peak_ongene[,2], "-", peak_ongene[,3])}
+    colnames(peak_ongene) <- c("seqnames", "Pstart", "Pend", "gene_id", "peak_on_gene")
+  }
+  peaks_length <- apply(peak_ongene, 1, function(x) {
+    length(as.numeric(x[2]):as.numeric(x[3]))
+  })
+  peak_ongene<- cbind(peak_ongene, peaks_length=peaks_length)
```

Comment:
This should be moved back to the condition

Reply:
Would it be an improvement to return the peak length in both cases (for `TSSmode` `T` or `F`)?

Reply:
only for TSS mode = T. The output for TSS mode =F is required for the activity computations

Reply:
Got it. I added a comment, so that I don't have this idea again